plugin.go

```diff
@@ -0,0 +1,168 @@
+)
+
+const (
+	rootPluginName         string        = "http"
+	pluginName             string        = "sendremotefile"
+	responseContentTypeKey string        = "Content-Type"
+	responseContentTypeVal string        = "application/octet-stream"
+	responseStatusCode     int           = http.StatusOK
+	xSendRemoteHeader      string        = "X-Sendremotefile"
+	bufferSize             int           = 10 * 1024 * 1024 // 10MB chunks
+	timeout                time.Duration = 5 * time.Second
+)
+
+type Configurer interface {
+	UnmarshalKey(name string, out any) error
+	Has(name string) bool
+}
+
+type Logger interface {
+	NamedLogger(name string) *zap.Logger
+}
+
+type Plugin struct {
+	log       *zap.Logger
+	bytesPool *sync.Pool
+}
+
+func (p *Plugin) Init(cfg Configurer, log Logger) error {
+	const op = rrErrors.Op("sendremotefile_plugin_init")
+
+	if !cfg.Has(rootPluginName) {
+		return rrErrors.E(op, rrErrors.Disabled)
+	}
+
+	p.log = log.NamedLogger(pluginName)
+	p.bytesPool = &sync.Pool{
+		New: func() interface{} {
+			b := make([]byte, bufferSize)
+			return &b
+		},
+	}
+
+	return nil
+}
+
+func (p *Plugin) Middleware(next http.Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		rec := httptest.NewRecorder()
+
+		next.ServeHTTP(rec, r)
+
+		// if there is no X-Sendremotefile header from the PHP worker, just return
+		if url := rec.Header().Get(xSendRemoteHeader); url == "" {
+			// re-add original headers, status code and body
+			maps.Copy(w.Header(), rec.Header())
+			w.WriteHeader(rec.Code)
+			_, err := w.Write(rec.Body.Bytes())
+			if err != nil {
+				p.log.Error("failed to write data to the response", zap.Error(err))
+			}
+
+			return
+		}
+
+		// we already checked that that header exists
+		url := rec.Header().Get(xSendRemoteHeader)
+		// delete the original X-Sendremotefile header
+		rec.Header().Del(xSendRemoteHeader)
+
+		if !strings.HasPrefix(url, "http") {
+			p.log.Error("header value must start with http")
+			http.Error(w, http.StatusText(http.StatusNotFound), http.StatusNotFound)
+			return
+		}
+
+		resp, err := NewClient(url, timeout).Request()
+		if err != nil {
+			var netErr net.Error
+			if errors.As(err, &netErr) && netErr.Timeout() {
+				http.Error(w, http.StatusText(http.StatusRequestTimeout), http.StatusRequestTimeout)
+				return
+			}
+
+			p.log.Error("failed to request from the upstream", zap.Error(err))
+			http.Error(w, http.StatusText(http.StatusInternalServerError), http.StatusInternalServerError)
+			return
+		}
+
+		defer func() {
+			err = resp.Body.Close()
+			if err != nil {
+				p.log.Error("failed to close upstream response body", zap.Error(err))
+			}
+		}()
+
+		if resp.StatusCode != http.StatusOK {
+			p.log.Error("invalid upstream response status code", zap.Int("status_code", resp.StatusCode))
+			http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+			return
+		}
+
+		bufPtr := p.bytesPool.Get().(*[]byte)
```

Comment:
You can have a look at [`bpool.go`](https://github.com/roadrunner-server/goridge/blob/master/internal/bpool.go) + [usage](https://github.com/roadrunner-server/goridge/blob/master/internal/receive.go#L83) to better implement pool with bytes slice. Also, no need to use `defer` since the only one place where it can be fired is the end of the method. So, it would be better to just put it here, since it adds overhead.